### PR TITLE
Data object

### PR DESCRIPTION
**Pull request by mvtnhan**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -12,134 +12,135 @@ import imgLoading from "./images/isLoading.gif";
 
 class App extends React.Component {
   state = {
-    todos: [],
+    todos: {
+      0: {
```
Comment (nghiamvt) [inline src/App.js:16]: id here should be Date right?

File: src/App.js [hunk 2]
```diff
@@ -12,134 +12,135 @@ import imgLoading from "./images/isLoading.gif";
...
+    const willTodo = Object.keys(todos).reduce(() => {
+      return Object.assign(
+        {},
+        Object.keys(todos)
+          .map((key) => todos[key])
+          .filter((todo) => todo.id !== id)
+      );
+    }, {});
+
+    //dungf filrer
```
Comment (nghiamvt) [inline src/App.js]: remove logs

File: src/App.js [hunk 3]
```diff
@@ -12,134 +12,135 @@ import imgLoading from "./images/isLoading.gif";
...
   };
 
   deleteTodo = (id) => {
-    axios.delete(`${URL.TODOS}/${id}`).then(() => {
-      this.setState({
-        todos: this.state.todos.filter((todo) => todo.id !== id),
-      });
+    //dung reduce
+    const { todos } = this.state;
+    const willTodo = Object.keys(todos).reduce(() => {
```
Comment (nghiamvt) [inline src/App.js]: what is willTodo? rename it

File: src/App.js [hunk 4]
```diff
@@ -12,134 +12,135 @@ import imgLoading from "./images/isLoading.gif";
...
+    //   .filter((todo) => todo.id !== id);
+    this.setState({
+      todos: Object.assign({}, willTodo),
     });
   };
 
-  updateStatus = (status) => {
+  clearCompleted = () => {
+    const { todos } = this.state;
+    const newTodo = Object.keys(todos).reduce(() => {
```
Comment (nghiamvt) [inline src/App.js]: does this function work?
check this again, I don't think it work

File: src/App.js [hunk 5]
```diff
@@ -12,134 +12,135 @@ import imgLoading from "./images/isLoading.gif";
...
-      })
-      .then(() => {
-        this.setState({
-          todos: this.state.todos.map((todo) => ({
-            ...todo,
-            done: todo.id === id ? !todo.done : todo.done,
-          })),
-        });
-      });
+  toggleTodo = (id, content, done) => {
```
Comment (nghiamvt) [inline src/App.js]: why doesn't this change to object like editTodo?
Did I tell u abt this? How about other places?

File: src/App.js (hunk 1 above)
Comment (mvtnhan) [reply src/App.js:16]: yes, id = Date.now()

File: src/App.js (hunk 2 above)
Comment (mvtnhan) [reply src/App.js]: ok

File: src/App.js (hunk 3 above)
Comment (mvtnhan) [reply src/App.js]: ok

File: src/App.js (hunk 4 above)
Comment (mvtnhan) [reply src/App.js]: it works

File: src/App.js (hunk 5 above)
Comment (mvtnhan) [reply src/App.js]: i did

File: src/App.js [hunk 6]
```diff
@@ -94,14 +94,13 @@ class App extends React.Component {
...
-          .filter((todo) => todo.id !== id)
-      );
-    }, {});
+    const newTodos = Object.keys(todos)
+      .map((key) => todos[key])
+      .filter((todo) => todo.id !== id)
+      .reduce((acc, item) => {
+        return Object.assign({}, acc, { [item.id]: item });
+      }, {});
+    console.log("newTodos", newTodos);
```
Comment (nghiamvt) [inline src/App.js]:
```
const newTodos = Object.keys(todos).reduce((acc, key) => {
    if (key === id) return acc;
    return {...acc, {[key]: todos[key]}
})
```

**Pull request closed without merging**